arangod/RocksDBEngine/RocksDBEdgeIndex.cpp

```diff
@@ -971,3 +1070,90 @@ void RocksDBEdgeIndex::recalculateEstimates() {
   }
   _estimator->setAppliedSeq(seq);
 }
+
+std::string_view RocksDBEdgeIndex::buildCompressedCacheKey(
+    std::string const*& previous, std::string_view value) const {
+  return _cacheKeyCollectionName.buildCompressedValue(previous, value);
+}
+
+std::string_view RocksDBEdgeIndex::buildCompressedCacheValue(
+    std::string const*& previous, std::string_view value) const {
+  return _cacheValueCollectionName.buildCompressedValue(previous, value);
+}
+
+std::string const* RocksDBEdgeIndex::cachedValueCollection(
+    std::string const*& previous) const noexcept {
+  if (previous == nullptr) {
+    previous = _cacheValueCollectionName.get();
+  }
+  TRI_ASSERT(previous != nullptr);
+  return previous;
+}
+
+RocksDBEdgeIndex::CachedCollectionName::CachedCollectionName() noexcept
+    : _name(nullptr) {}
+
+RocksDBEdgeIndex::CachedCollectionName::~CachedCollectionName() {
+  delete _name.load(std::memory_order_relaxed);
```

Comment:
One thread create string, another destroy string, formally you need happens before (maybe it's done by another synchronization) and in practice it doesn't really matter.
But I believe honest acquire will be better for this, because it's doesn't look like performance critical.

Reply:
Hmm
> maybe it's done by another synchronization

This probably should always to be true, because otherwise we don't have hb between dtor and calling method